remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java

```diff
@@ -56,22 +50,17 @@
 import org.infinispan.query.remote.impl.logging.Log;
 import org.infinispan.query.remote.impl.persistence.PersistenceContextInitializerImpl;
 import org.infinispan.registry.InternalCacheRegistry;
-import org.infinispan.security.AuthorizationPermission;
-import org.infinispan.security.impl.CacheRoleImpl;
 import org.infinispan.server.core.dataconversion.ProtostreamJsonTranscoder;
 import org.infinispan.server.core.dataconversion.ProtostreamObjectTranscoder;
 import org.infinispan.server.core.dataconversion.ProtostreamTextTranscoder;
-import org.infinispan.transaction.LockingMode;
-import org.infinispan.transaction.TransactionMode;
-import org.infinispan.util.concurrent.IsolationLevel;
 
 /**
  * Initializes components for remote query. Each cache manager has its own instance of this class during its lifetime.
  *
  * @author anistor@redhat.com
  * @since 6.0
  */
-@InfinispanModule(name = "remote-query-server", requiredModules = "core")
+@InfinispanModule(name = "remote-query-server", requiredModules = {"core", "query"})
```

Comment:
Should we depend on "server-core" instead of just "core"?

Reply:
No, we just use core services (component registry, cache, cache manager). The "server-core" might be the overarching module depending on "remote-query-server". Currently it's not.

Reply:
Ok, makes sense. I was thinking because we had a maven dependency on server-core, it would also make sense to have one here, but I understand I was thinking about it in the wrong way.

You will be interested in https://github.com/infinispan/infinispan/pull/7391 as it introduces a dependency on the server-core module initialization (in it's current form at least). This is because the remote-query module overrides the SerializationContext of the protostream transcoder that is registered in the server-core module. Hence we need a happens before relationship.

Reply:
Until now we did not use components from server-core, so even though we have a maven dependency on infinispan-server-core (for some json stuff and dataconversion) that did not need to be mirrored into a component registry dependency.  But with your new changes we can add that if needed.